scripts/combine_control_files.py

```diff
@@ -45,11 +45,18 @@
     BUILD_TAG = args.BUILD_TAG
     public_ext_list = args.public_extensions.split(",")
 
+    build_tags = {}
+    with open("build_tags.txt", "r") as f:
+        for line in f.readlines():
```

Comment:
`readlines` will read the whole file at once. Did you do it intentionally?

```python
for line in f:
```

in that case, it will be reading line by line

Reply:
fixed